### ISSUE TITLE
parsing config files

### ISSUE THREAD

**Issue by iamlambda** [adamharmansky/showkeys#1]
Instead of getting keybinds from config files you write by hand, it would be better to parse specific config files according to each program.
btw [this](https://github.com/adamharmansky/showkeys/blob/97bdeb5bd87ead41ac64b6e5598b76604fce2bf4/showkeys.c#L128) is very pog.

**Comment by adamharmansky** [comment]
Yes, I am currently working on the loading mechanism. I will need to split it into multiple functions, like load_program, load_category and load_key and optimize the string operations.

I probably don't want to parse the individual config files in the program itself, but rather create a script that will convert the files.

**Comment by adamharmansky** [comment]
Alright, the load function is fixed.

I will try to make a script that parses dwm `config.h`, we'll see how reliable it is.

I'll leave this thread open if anyone has suggestions on parsing config files.

**Comment by 0Karakurt0** [comment]
I'd say it would be more beneficial to make showkeys read it's own config and provide a bunch of scripts that would generate this config from other programs configs (dwm, i3wm, vim, etc).
Besides... Why directories with text files? I was expecting something like JSON
